packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -0,0 +1,519 @@
+  /**
+   * 64 Aurora Capacity Units
+   */
+  ACU_64 = 64,
+  /**
+   * 128 Aurora Capacity Units
+   */
+  ACU_128 = 128,
+  /**
+   * 192 Aurora Capacity Units
+   */
+  ACU_192 = 192,
+  /**
+   * 256 Aurora Capacity Units
+   */
+  ACU_256 = 256,
+  /**
+   * 384 Aurora Capacity Units
+   */
+  ACU_384 = 384
+}
+
+/**
+ * Options for configuring scaling on an Aurora Serverless cluster
+ */
+export interface ServerlessScalingOptions {
+  /**
+   * The minimum capacity for an Aurora Serverless database cluster.
+   *
+   * @default - determined by Aurora based on database engine
+   */
+  readonly minCapacity?: AuroraCapacityUnit;
+
+  /**
+   * The maximum capacity for an Aurora Serverless database cluster.
+   *
+   * @default - determined by Aurora based on database engine
+   */
+  readonly maxCapacity?: AuroraCapacityUnit;
+
+  /**
+   * The time before an Aurora Serverless database cluster is paused.
+   * A database cluster can be paused only when it is idle (it has no connections).
+   *
+   * If a DB cluster is paused for more than seven days, the DB cluster might be
+   * backed up with a snapshot. In this case, the DB cluster is restored when there
+   * is a request to connect to it.
+   *
+   * Set to 0 to disable
+   *
+   * @default - automatic pause enabled after 5 minutes
+   */
+  readonly autoPause?: Duration;
+}
+
+/**
+ * New or imported Serverless Cluster
+ */
+abstract class ServerlessClusterBase extends Resource implements IServerlessCluster {
+  /**
+   * Identifier of the cluster
+   */
+  public abstract readonly clusterIdentifier: string;
+
+  /**
+   * The endpoint to use for read/write operations
+   */
+  public abstract readonly clusterEndpoint: Endpoint;
+
+  /**
+   * The endpoint to use for read/write operations
+   */
+  public abstract readonly clusterReadEndpoint: Endpoint;
+
+  /**
+   * Access to the network connections
+   */
+  public abstract readonly connections: ec2.Connections;
+
+  /**
+   * Renders the secret attachment target specifications.
+   */
+  public asSecretAttachmentTarget(): secretsmanager.SecretAttachmentTargetProps {
+    return {
+      targetId: this.clusterIdentifier,
+      targetType: secretsmanager.AttachmentTargetType.RDS_DB_CLUSTER,
+    };
+  }
+}
+
+/**
+ * Create an Aurora Serverless Cluster
+ *
+ * @resource AWS::RDS::DBCluster
+ */
+export class ServerlessCluster extends ServerlessClusterBase {
+
+  /**
+   * Import an existing DatabaseCluster from properties
+   */
+  public static fromServerlessClusterAttributes(scope: Construct, id: string,
+    attrs: ServerlessClusterAttributes): IServerlessCluster {
+
+    return new ImportedServerlessCluster(scope, id, attrs);
+  }
+
+  public readonly clusterIdentifier: string;
+  public readonly clusterEndpoint: Endpoint;
+  public readonly clusterReadEndpoint: Endpoint;
+  public readonly connections: ec2.Connections;
+
+  /**
+   * The secret attached to this cluster
+   */
+  public readonly secret?: secretsmanager.ISecret;
```

Comment:
Any reason this is `public`?

Reply:
was doing this to stay in concert with `cluster` and `instance` where they're also public... I think private makes sense, as does consistency. thoughts?

Reply:
Let's start with `private` - we can always make it `public` later (but not the other way round 🙂).

Reply:
done

Reply:
:/ - Missed this comment. The `secret` is public so users can log in after. Without this, users who do the default and allow the password to be auto-generated don't have a programmatic way to get at that secret later (e.g., reference in their compute infrastructure).

Reply:
thanks Nick, caught up on usage with the Data API just now and approved the related PR to fix.